pycbc/strain/recalibrate.py

```diff
@@ -512,3 +515,109 @@ def map_to_adjust(self, strain, **params):
                               kappa_pu_re=calib_args[6],
                               kappa_pu_im=calib_args[7])
         return strain_adjusted
+
+
+def read_calibration_envelop_file(calibration_file, correction_type,
+                                  minimum_frequency, maximum_frequency, n_nodes):
+    """
+    This function reads the calibration envelop file and provide arrays
+    needed to construct cubic splines
+
+    Input:
+      calibration_file: Calibration envelop file
+      correction_type: Specify the correction type between 'data' and 'template'
+      minimum_frequency: Lower cut-off frequency to apply calibration
+      maximum_frequency: Upper cut-off frequency to apply calibration
+      n_nodes: Nodal points for cubic-splines
+
+    Output:
+      log_nodes: Values of log-frequency for nodal points
+      amplitude_median_nodes: Median values of Amplitude correction at frequency nodal points
+      amplitude_sigma_nodes: Standard deviation values of Amplitude correction at frequency nodal points
+      phase_median_nodes: Median values of Phase correction at frequency nodal points
+      phase_sigma_nodes: Standard deviation values of Phase correction at frequency nodal points
+    """
+    calibration_data = np.loadtxt(calibration_file).T
+    log_frequency_array = np.log(calibration_data[0])
+
+    log_nodes = np.linspace(np.log(minimum_frequency),
+                            np.log(maximum_frequency), n_nodes)
+
+    if correction_type.lower()=='template':
+        amplitude_median = calibration_data[1] - 1
+        phase_median = calibration_data[2]
+        amplitude_sigma = abs(calibration_data[5] - calibration_data[3]) / 2
+        phase_sigma = abs(calibration_data[6] - calibration_data[4]) / 2
+
+    elif correction_type.lower()=='data':
+        amplitude_median = 1/calibration_data[1] -1
+        phase_median = -calibration_data[2]
+        amplitude_sigma = abs(1/calibration_data[3] - 1/calibration_data[5]) / 2
+        phase_sigma = abs(calibration_data[6] - calibration_data[4]) / 2
```

Comment:
I'd suggest adding an `else` block to this if/else statement and raising an error since the `calibration_type` is an invalid value.

Reply:
Sounds good. Will do it in following commit.